docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+### Non-Goals
+This proposal doesn’t define how either the client or the server must use this data, though we make some non-normative recommendations (e.g. “this data may be displayed to the user” or “a suitable backend will be able to translate this to a fully qualified URL”).
+
+
+Consequently, this document does not describe the implementation of either client or server sides of this spec. While it defines the pieces necessary for developing and deploying a chart, it does not define a development workflow. Some considerations are listed in [Appendix B](#appendix-b-the-standard-chart-repository).
+
+
+## The Chart Format
+Borrowing the nomenclature from Helm, we define a Chart as a bundle of data and metadata whose contents are sufficient for installing a workload into a Kubernetes cluster running DM.
+
+
+A Chart is composed of the following pieces:
+
+
+1. A human- and machine-readable metadata file called `Chart.yaml`
+2. A directory named `templates/`
+3. An optional `README.md` file
+4. An optional `LICENSE` file
+5. An optional `docs/` directory
+6. An optional `hooks/` directory
+7. An optional `icon.svg.`
+
+
+A chart _may_ have other directories and files, but such files are not defined by or required by this proposal. For the purposes of this proposal, any additional files do not contribute to the core functionality of chart installation.
+
+
+_Note: When a chart file is deployed, a [provenance file](#the-provenance-file) is generated for the chart. That file is not stored inside of the chart, but is considered part of the chart’s packaged format._
+
+
+The chart.yaml file format is described [later in this document](#the-chart-file).
+
+
+The `templates/` directory contains one or more template files, as defined in the [Directory Layout](#directory-layout) section. Templates, along with all of their supporting files, are located therein.
+
+
+An optional `README.md` file may be specified, which contains long-form text information about using this chart. Tools may display this information, if present. The README.md file is in Markdown format, and should contain information about a Chart’s purpose, usage, and development.
+
+
+An optional `LICENSE` file may be present, which specifies license information for this chart and/or the images dependent on it.
+
+
+An optional `docs/` directory may be present, and may contain one or more documentation files. This directory contains documentation that is more specific, verbose, or thorough than what is present in the `README.md` file.
+
+
+An optional `hooks/` directory may be present. **This is reserved for future use.** A deployment engine may specify lifecycle hooks, such as `pre-install`, `pre-render`, `post-install` and `post-render`. Lifecycle hooks provide an opportunity for a chart author to run additional steps in the processing pipeline. Implementations of these hooks are stored here. _No other files may be stored in this directory._
```

Comment:
That's a good idea. Let's talk about it in the future. We just figured this would be a good placeholder for now. The intent is to reserve the directory so no one would use it. Would that be okay with you? @jackgr